vendor/github.com/docker/libnetwork/network.go

```diff
@@ -1181,7 +1181,8 @@ func (n *network) createEndpoint(name string, options ...EndpointOption) (Endpoi
 	ep.locator = n.getController().clusterHostID()
 	ep.network, err = ep.getNetworkFromStore()
 	if err != nil {
-		return nil, fmt.Errorf("failed to get network during CreateEndpoint: %v", err)
+		logrus.Errorf("failed to get network during CreateEndpoint: %v", err)
+		return nil, ErrNoSuchNetwork(n.Name())
```

Comment:
I'd prefer `ep.getNetworkFromStore()` to return this error-type.

While it may be true _currently_ that this is always due to the network not being found, it may not be in future. I don't think we should make the assumption here that this is a `ErrNoSuchNetwork`.

Reply:
Yes, we can check the err of `ep.getNetworkFromStore()` and return `ErrNoSuchNetwork` only when the error is really network-not-found.

Reply:
I have reverted the code change in libnetwork. Let's continue work on libnetwork project.